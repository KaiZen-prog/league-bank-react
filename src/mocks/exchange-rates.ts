export const exchangeRates = {
  '2024-01-19': {
    USD: 1,
    RUB: 88.573959,
    EUR: 0.918438,
    KZT: 452.633544,
    GBP: 0.787512,
    CNY: 7.1197
  },
  '2024-01-18': {
    USD: 1,
    RUB: 88.638497,
    EUR: 0.918649,
    KZT: 453.345273,
    GBP: 0.788835,
    CNY: 7.1263
  },
  '2024-01-17': {
    USD: 1,
    RUB: 88.105727,
    EUR: 0.919239,
    KZT: 453.139857,
    GBP: 0.791321,
    CNY: 7.1265
  },
  '2024-01-16': {
    USD: 1,
    RUB: 87.719298,
    EUR: 0.913662,
    KZT: 451.339299,
    GBP: 0.786436,
    CNY: 7.1073
  },
  '2024-01-15': {
    USD: 1,
    RUB: 88.043653,
    EUR: 0.913639,
    KZT: 451.160232,
    GBP: 0.785191,
    CNY: 7.111
  },
  '2024-01-14': {
    USD: 1,
    RUB: 87.959997,
    EUR: 0.912076,
    KZT: 451.160232,
    GBP: 0.78456,
    CNY: 7.1126
  },
  '2024-01-13': {
    USD: 1,
    RUB: 87.959997,
    EUR: 0.912076,
    KZT: 451.530905,
    GBP: 0.78456,
    CNY: 7.1106
  },
  '2024-01-12': {
    USD: 1,
    RUB: 88.495575,
    EUR: 0.910544,
    KZT: 451.995747,
    GBP: 0.782515,
    CNY: 7.09752
  }
};
