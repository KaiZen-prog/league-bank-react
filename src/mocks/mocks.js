export const promoSlides = [
  `img/desktop-slide-1.jpg`,
  `img/desktop-slide-2.jpg`,
  `img/desktop-slide-3.jpg`,
];
