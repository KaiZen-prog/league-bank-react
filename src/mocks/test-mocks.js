import moment from "moment";

export const InitialState = {
  CONVERSION_HISTORY: {
    conversionHistory: [
      {
        date: `18.09.2021`,
        currencyInput: {
          type: `USD`,
          amount: 6549
        },
        currencyOutput: {
          type: `EUR`,
          amount: 5585.07
        }
      },
      {
        date: `19.09.2021`,
        currencyInput: {
          type: `CNY`,
          amount: 23
        },
        currencyOutput: {
          type: `RUB`,
          amount: 258.59
        }
      },
      {
        date: `20.09.2021`,
        currencyInput: {
          type: `EUR`,
          amount: 12458
        },
        currencyOutput: {
          type: `GBP`,
          amount: 10694.16
        }
      },
      {
        date: `13.09.2021`,
        currencyInput: {
          type: `CNY`,
          amount: 654
        },
        currencyOutput: {
          type: `RUB`,
          amount: 7363.93
        }
      },
      {
        date: `15.09.2021`,
        currencyInput: {
          type: `EUR`,
          amount: 105
        },
        currencyOutput: {
          type: `GBP`,
          amount: 89.6
        }
      },
      {
        date: `20.09.2021`,
        currencyInput: {
          type: `RUB`,
          amount: 105
        },
        currencyOutput: {
          type: `USD`,
          amount: 1.43
        }
      }
    ]
  },

  CONVERTER: {
    date: moment().utc().format(`YYYY-MM-DD`),
    exchangeRate: {
      USD: 1,
      RUB: 72.8797,
      EUR: 0.852813,
      GBP: 0.727776,
      CNY: 6.4662
    }
  }
};
