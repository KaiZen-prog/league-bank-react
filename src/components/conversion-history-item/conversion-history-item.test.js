import React from "react";
import renderer from "react-test-renderer";
import {Provider} from 'react-redux';
import {Router} from "react-router-dom";
import {createMemoryHistory} from "history";
import configureStore from 'redux-mock-store';
import ConversionHistoryItem from "./conversion-history-item";
import {InitialState} from "../../mocks/test-mocks";

const mockStore = configureStore();
const history = createMemoryHistory({initialEntries: [`/`]});

const noop = () => {};

test(`ConversionHistoryItem render correctly`, () => {
  const store = mockStore(InitialState);
  const props = InitialState.CONVERSION_HISTORY.conversionHistory[0];

  const tree = renderer
      .create(
          <Provider store={store}>
            <Router history={history}>
              <ConversionHistoryItem
                date={props.date}
                inputAmount={props.currencyInput.amount}
                inputCurrency={props.currencyInput.type}
                outputAmount={props.currencyOutput.amount}
                outputCurrency={props.currencyInput.type}
                clear={noop}
              />
            </Router>
          </Provider>
      ).toJSON();
  expect(tree).toMatchSnapshot();
});
