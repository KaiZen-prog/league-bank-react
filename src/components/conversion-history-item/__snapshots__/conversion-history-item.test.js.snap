// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversionHistoryItem render correctly 1`] = `
<li
  className="conversion-history__item"
>
  <p
    className="conversion-history__date"
  >
    18.09.2021
  </p>
  <div
    className="conversion-history__container"
  >
    <p
      className="conversion-history__before"
    >
      6549
       
      USD
    </p>
    <p
      className="conversion-history__after"
    >
      5585,07
       
      USD
    </p>
  </div>
</li>
`;
